Refactoring

Flre-fly opened the pull request:
### keword 
- KEYWORD #{24}

### summary
- 회의때 결정된 내용 반영하기

### as-is / to-be
- 회의에서 결정된 내용이랑 디비설계가 달랐는데 이를 맞춰줌

### review point
- 체크리스트 잘 완수했는지 정도만 봐주시면 돼요

Flre-fly committed:
backend-24 코드 리팩토링
feat: member 컬럼으로 office추가
: s3 toast ui 를 연동하기 위해 index.html추가(추후에 controller등 수정 예정)

Flre-fly committed:
backend-24 코드 리팩토링
refactoring: member 컬럼으로 office추가하고 오류나는 부분 수정
feat : demo entity추가
: history entity추가
: project에 content 컬럼 추가

Flre-fly committed:
backend-24 코드 리팩토링
feat : demo, project에 participants 컬럼 추가
: demo, project, thesis와 field의 일대일 양방향 관계 추가
: field repository에 demo, project, thesis별로 찾아오는 메서드 추가
refactoring : researcher의 research 컬럼 제거(fields로 쓸테니까)
: admission을 string이 아닌 localdatetime타입으로 수정
: field에 name 컬럼추가하고 Category제거

Flre-fly committed:
backend-24 코드 리팩토링
conflict수정

Flre-fly committed:
backend-24 코드 리팩토링
: admin과 member합치기

Byuntil commented on src/main/java/byuntil/backend/research/service/DemoService.java:
```diff
@@ -0,0 +1,38 @@
+    public void deleteById(Long id){
+        demoRepository.deleteById(id);
+    }
+    public void update(DemoDto demoDto){
```
update 이 부분 일부러 비워둔거야?

Flre-fly replied:
아니 빼먹었다.. 다시 작성해서 commit했어

Flre-fly committed:
backend-24 코드 리팩토링
feat : demo, field, project, thesis에 update메서드 추가
test : 삭제시 예외가 일어나는 경우, 잘 삭제되는 두가지 경우 test메서드 추가
